stratosphere/pm/source/impl/pm_process_manager.cpp

```diff
@@ -0,0 +1,680 @@
+        {
+            u64 tmp = 0;
+            R_ASSERT(svcGetProcessInfo(&tmp, process_info->GetHandle(), ProcessInfoType_ProcessState));
+            process_info->SetState(static_cast<ProcessState>(tmp));
+        }
+        const ProcessState new_state = process_info->GetState();
+
+        /* If we're transitioning away from crashed, clear waiting attached. */
+        if (old_state == ProcessState_Crashed && new_state != ProcessState_Crashed) {
+            process_info->ClearExceptionWaitingAttach();
+        }
+
+        switch (new_state) {
+            case ProcessState_Created:
+            case ProcessState_CreatedAttached:
+            case ProcessState_Exiting:
+                break;
+            case ProcessState_Running:
+                if (process_info->ShouldSignalOnDebugEvent()) {
+                    process_info->ClearSuspended();
+                    process_info->SetSuspendedStateChanged();
+                    g_process_event->Signal();
+                } else if (GetRuntimeFirmwareVersion() >= FirmwareVersion_200 && process_info->ShouldSignalOnStart()) {
+                    process_info->SetStartedStateChanged();
+                    process_info->ClearSignalOnStart();
+                    g_process_event->Signal();
+                }
+                break;
+            case ProcessState_Crashed:
+                process_info->SetExceptionOccurred();
+                g_process_event->Signal();
+                break;
+            case ProcessState_RunningAttached:
+                if (process_info->ShouldSignalOnDebugEvent()) {
+                    process_info->ClearSuspended();
+                    process_info->SetSuspendedStateChanged();
+                    g_process_event->Signal();
+                }
+                break;
+            case ProcessState_Exited:
+                if (GetRuntimeFirmwareVersion() < FirmwareVersion_500 && process_info->ShouldSignalOnExit()) {
+                    g_process_event->Signal();
+                } else {
+                    ProcessListAccessor list(g_process_list);
+                    CleanupProcess(list, process_info);
+                }
+                /* Return ConnectionClosed to cause libstratosphere to stop waiting on the process. */
+                return ResultKernelConnectionClosed;
+            case ProcessState_DebugSuspended:
+                if (process_info->ShouldSignalOnDebugEvent()) {
+                    process_info->SetSuspended();
+                    process_info->SetSuspendedStateChanged();
+                    g_process_event->Signal();
+                }
+                break;
+        }
+
+        return ResultSuccess;
+    }
+
+    /* Process Management. */
+    Result LaunchTitle(u64 *out_process_id, const ncm::TitleLocation &loc, u32 flags) {
+        /* Ensure we only try to launch one title at a time. */
+        static HosMutex s_lock;
+        std::scoped_lock<HosMutex> lk(s_lock);
```

Comment:
```suggestion
        std::scoped_lock lk(s_lock);
```

Reply:
Wait can I really elide that? There's like a million places to fix this if so.

Reply:
Yep! Class template arguments can now be deduced starting from C++17 :)

Reply:
That's awesome, and good info to have.